src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -242,6 +245,13 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
             }
             else if (parameter.ParameterType == typeof(string) || TryParseMethodCache.HasTryParseMethod(parameter))
             {
+                // We're in the fallback case and we have a parameter and route parameter match so don't fallback
+                // to query string in this case
+                if (factoryContext.RouteParameters is { } routeParams && routeParams.Contains(parameter.Name))
+                {
+                    return BindParameterFromProperty(parameter, RouteValuesExpr, parameter.Name, factoryContext);
+                }
+
                 return BindParameterFromRouteValueOrQueryString(parameter, parameter.Name, factoryContext);
```

Comment:
Should this even attempt to bind from `RouteValues` anymore?

Reply:
The only scenario is if the list of routes is null. But maybe if it's not null and it's not a route, the we fallback directly to query? I can add that in rc1